cli/tests/test_status_command.rs

```diff
@@ -81,3 +81,58 @@ fn test_status_filtered() {
     Parent commit: zzzzzzzz 00000000 (empty) (no description set)
     "###);
 }
+
+// See <https://github.com/martinvonz/jj/issues/3108>
+#[test]
+fn test_status_display_rebase_instructions() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+
+    let repo_path = test_env.env_root().join("repo");
+    let conflicted_path = repo_path.join("conflicted.txt");
+
+    // PARENT: Write the initial file
+    std::fs::write(&conflicted_path, "initial contents").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["describe", "--message", "Initial contents"]);
+
+    // CHILD1: New commit on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["new", "--message", "First part of conflicting change"],
+    );
+    std::fs::write(&conflicted_path, "Child 1").unwrap();
+
+    // CHILD2: New commit also on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &[
+            "new",
+            "--message",
+            "Second part of conflicting change",
+            "@-",
+        ],
+    );
+    std::fs::write(&conflicted_path, "Child 2").unwrap();
+
+    // CONFLICT: New commit that is conflicted by merging <CHILD1> and <CHILD2>
+    test_env.jj_cmd_ok(&repo_path, &["new", "--message", "boom", "all:(@-)+"]);
+    // Adding more descendants to ensure we correctly find the root ancestors with
+    // conflicts, not just the parents.
+    test_env.jj_cmd_ok(&repo_path, &["new", "--message", "boom-cont"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "--message", "boom-cont-2"]);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["status"]);
+
+    insta::assert_snapshot!(stdout, @r###"
+    The working copy is clean
+    There are unresolved conflicts at these paths:
+    conflicted.txt    2-sided conflict
+    Working copy : yqosqzyt d41fb742 (conflict) (empty) boom-cont-2
+    Parent commit: royxmykx c2389cec (conflict) (empty) boom-cont
+    To resolve the conflicts, start by updating to the first one:
+      jj new mzvwutvlkqwt
```

Comment:
How do we know that this it the right commit? Could you add a `jj log` output above?

Reply:
Added an insta check of the log output to show the oldest commit with conflicts is the one named in status :)